sdcm/argus_test_run.py

```diff
@@ -235,9 +236,12 @@ def from_sct_config(cls, test_id: UUID, sct_config: SCTConfiguration) -> TestRun
         config_files = sct_config.get("config_files")
         started_by = get_username()
 
+        # start time is a part of primary key and getting same start time will cause an overwrite
+        safe_start_time = datetime.utcnow().replace(microsecond=0) + timedelta(seconds=(randint(10, 60)))
```

Comment:
it doesn't sound that safe to me... it's still can get into that situation cause it's random. 

why aren't we fixing this on the server side ? cause we need to change the schema ?

why are we zeroing the microseconds ? leaving them in sound a bit more safer...

Reply:
i agree that fixing it on the server side would have bigger benefit, but perhaps, because of kind of "low" parallelism in these jobs, it won't be a problem the current situation, until a proper fix is done in the server side...

Reply:
other way to mitigate it, is to delay the start of parallel jobs inside the same build by, let's say, 1 second, as it won't make much difference in the end result, and will most certainly mitigate this issue

Reply:
> it doesn't sound that safe to me... it's still can get into that situation cause it's random.

I agree, there is still a possibility of getting runs to random the same interval while also starting at the same time, or worse, getting a random interval on one run enough to intersect with start time of another run.

> why aren't we fixing this on the server side ? cause we need to change the schema ?

This will be fixed serverside (and in fact kinda is fixed already since timestamp is calculated serverside and is not submitted in the new client), but since sct is still interacting directly with db this is a temporary fix for that.

> why are we zeroing the microseconds ? leaving them in sound a bit more safer...

It seems that scylla (at least by default) doesn't accept microsecond precision in the timestamps, I'm zeroing them to resolve that, but if there's a way to make scylla to accept it that would be better.

Reply:
much better, 

also when do we plan to switch to the new client api ?